meson.build

```
@@ -0,0 +1,125 @@
+project(
+  'spoa',
+  ['cpp'],
+  version : '4.1.0',
+  default_options : [
+    'buildtype=release',
+    'warning_level=2',
+    'cpp_std=c++11',
+    'b_ndebug=if-release'],
+  license : 'MIT',
+  meson_version : '>=0.60.0'
+)
+
+############
+# CXXFLAGS #
+############
+
+spoa_warning_flags = []
+cpp = meson.get_compiler('cpp')
+foreach cflag : [
+  '-Wall',
+  '-Wextra',
+  '-pedantic']
+    if cpp.has_argument(cflag)
+      spoa_warning_flags += cflag
+    endif
+endforeach
+
+################
+# Dependencies #
+################
+
+spoa_lib_flags = []
+spoa_lib_deps = []
+
+if (get_option('spoa_use_cereal'))
```

```suggestion
if (get_option('cereal'))
```
I see you namespaced all spoa options, my question is: why? meson options are already namespaced, that is, `-Dbla` only triggers `bla` for the main project, not for the subprojects.